packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -5589,28 +5642,290 @@ export default class Meeting extends StatelessWebexPlugin {
+        name: 'client.ice.end',
+        payload: {
+          canProceed: false,
+          icePhase: 'JOIN_MEETING_FINAL',
+          errors: [
+            // @ts-ignore
+            this.webex.internal.newMetrics.callDiagnosticMetrics.getErrorPayloadForClientErrorCode({
+              clientErrorCode: CallDiagnosticUtils.generateClientErrorCodeForIceFailure({
+                signalingState:
+                  this.mediaProperties.webrtcMediaConnection?.multistreamConnection?.pc?.pc
+                    ?.signalingState ||
+                  this.mediaProperties.webrtcMediaConnection?.mediaConnection?.pc?.signalingState ||
+                  'unknown',
+                iceConnectionState:
+                  this.mediaProperties.webrtcMediaConnection?.multistreamConnection?.pc?.pc
+                    ?.iceConnectionState ||
+                  this.mediaProperties.webrtcMediaConnection?.mediaConnection?.pc
+                    ?.iceConnectionState ||
+                  'unknown',
+                turnServerUsed: this.turnServerUsed,
+              }),
+            }),
+          ],
+        },
+        options: {
+          meetingId: this.id,
+        },
+      });
+      throw new Error(
+        `Timed out waiting for media connection to be connected, correlationId=${this.correlationId}`
+      );
+    }
+  }
+
+  /**
+   * Enables statsAnalyser if config allows it
+   *
+   * @private
+   * @returns {void}
+   */
+  private createStatsAnalyzer() {
+    // @ts-ignore - config coming from registerPlugin
+    if (this.config.stats.enableStatsAnalyzer) {
+      // @ts-ignore - config coming from registerPlugin
+      this.networkQualityMonitor = new NetworkQualityMonitor(this.config.stats);
+      this.statsAnalyzer = new StatsAnalyzer(
+        // @ts-ignore - config coming from registerPlugin
+        this.config.stats,
+        (ssrc: number) => this.receiveSlotManager.findReceiveSlotBySsrc(ssrc),
+        this.networkQualityMonitor
+      );
+      this.setupStatsAnalyzerEventHandlers();
+      this.networkQualityMonitor.on(
+        EVENT_TRIGGERS.NETWORK_QUALITY,
+        this.sendNetworkQualityEvent.bind(this)
+      );
+    }
+  }
+
+  /**
+   * Handles device logging
+   *
+   * @private
+   * @static
+   * @returns {Promise<void>}
+   */
+  private static async handleDeviceLogging(): Promise<void> {
+    try {
+      const devices = await getDevices();
+
+      MeetingUtil.handleDeviceLogging(devices);
+    } catch {
+      // getDevices may fail if we don't have browser permissions, that's ok, we still can have a media connection
+    }
+  }
+
+  /**
+   * Returns a promise. This promise is created once the local sdp offer has been successfully created and is resolved
+   * once the remote sdp answer has been received.
+   *
+   * @private
+   * @returns {Promise<void>}
+   */
+  private async waitForRemoteSDPAnswer(): Promise<void> {
+    const LOG_HEADER = 'Meeting:index#addMedia():waitForRemoteSDPAnswer -->';
```

Comment:
I'm not so sure I'm a fan of using the addMedia() as the log header for each of these functions. I think it's still an improvement on whats there now, but it does make it tricky in the future.

Reply:
I think it's important to have a common string in all the functions that get called from addMedia() - this makes it very easy to search the logs for addMedia() related issues and these functions are not called from anywhere else anyway, if any of them does get called from other place then that "addMedia" string can be passed as an argument